Fix for bug in PHP FPM through both ProxyPassMatch and SetHandler

Caffe1neAdd1ct opened the pull request:
where the global variable SERVER SCRIPT_NAME isn't set correctly on through apache 2.4 to php-fpm.

fixes #7719

MorrisJobke committed:
Add expiration date to share by link view

MorrisJobke committed:
add "Never" if expiration date is not set

MorrisJobke committed:
Not set expiration date is represented as 0

MorrisJobke committed:
Update header during update callback

rullzer committed:
Group shares with same source and target

Fixes #24575

Note that this is a very limited solution and eventually we want smarter
merging!

Henni committed:
remove ie8+9 support

icewind1991 committed:
Add option to `occ files_external:list` to show all configured mounts

Commit:
Merge pull request #25225 from owncloud/enc-revertversionsize

Rollback version must also adjust cached size

Commit:
Init the headers in iedavclient.js

MorrisJobke committed:
Merge pull request #25233 from owncloud/iedavclient-nullheaders

Init the headers in iedavclient.js

Commit:
[tx-robot] updated from transifex

Commit:
Prerender file list pages to include search results

When filtering the file list, if a result is on an unrendered page,
make sure to call _nextPage() to prerender the pages in order to
display all matching results.

Commit:
Fix trashed file name corruption when rerendering trashbin list

The trashbin code does some gymnastics with the file data at render
time. This fix makes sure that the modifications done on the file data
are only local to the rendering code and doesn't affect the actual file
data from the file list.

Commit:
Don't reload page in case of auth errors during setup checks

If an error occurs during setup checks, do not let the global ajax
error handler reload the page.

Henni committed:
remove javascript png fallback for svg

Henni committed:
remove png actions, fileicons, etc

Henni committed:
remove png references in core

Henni committed:
remove svg classes

icewind1991 committed:
Remove a fed share from the local table before trying to notify the remote server

Commit:
Merge pull request #25238 from owncloud/files-trashlistnamecorruption

Fix trashed file name corruption when rerendering trashbin list

Commit:
Add explicit delete permission to link shares

Link shares always allowed deletion, however internally the permissions
were stored as 7 which lacked delete permissions. This created an
inconsistency in the Webdav permissions.

This fix makes sure we include delete permissions in the share
permissions, which now become 15.

In case a client is still passing 7 for legacy reasons, it gets
converted automatically to 15.

Commit:
Repair step to adjust link share delete permissions

Commit:
Use OC.Backbone instead of Backbone directly in authtoken JS code

Fixes asset pipeline issue with the auth token in personal page

Commit:
Remove tooltip when disconnecting token

Commit:
Fix "Other" value handling in quota dropdown in users page

Prevents "other" value to be deleted.
When appending custom value, put it above the "other" entry.

Commit:
[tx-robot] updated from transifex

Commit:
Prevent negative or non-number values in quota input

Commit:
Merge pull request #25239 from owncloud/setupchecks-preventreload

Don't reload page in case of auth errors during setup checks

Commit:
Show error message when posting an invalid comment

When an internal server error occurs while creating or updating a
comment, display a proper error notification in the UI.

Commit:
Merge pull request #25171 from owncloud/files_external-list-all

Add option to `occ files_external:list` to show all configured mounts

Commit:
Make code integrity check work when OC is not installed yet

Commit:
Fix search result link for file results outside default list

When outside the "All files" list, the search result link must properly
redirect to the "All files" list.

Commit:
Hide search results after switching directory

When clicking on a folder result in the search result list, the result
box for "results in another folder" must disappear.

ChristophWurst committed:
check login name when authenticating with client token

ChristophWurst committed:
show which login name to use for the new app password

icewind1991 committed:
better detect errors in fed sharing response

icewind1991 committed:
Handle exceptions thrown while trying to notify remote server of a fed share

Commit:
Keep encryption enabled if decrypting for single user

When decrypting all files of a single user, the admin usually does not
intend encryption to be suddenly disabled for everyone. This fix
reenables encryption after decrypting for a single user.

Decrypting for all users will still disable encryption globally.

ChristophWurst committed:
fix layout

ChristophWurst committed:
Merge pull request #25257 from owncloud/comments-showerroronsave

Show error message when posting an invalid comment

ChristophWurst committed:
Merge pull request #25259 from owncloud/search-fixsearchfromotherfilelists

Fix search result link for file results outside default list

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
Merge pull request #25154 from owncloud/token-login-check-loginname

check login name when authenticating via token and basic auth

Commit:
Merge pull request #25251 from owncloud/authtoken-assetpipelinefix

Use OC.Backbone instead of Backbone directly in authtoken JS code

Commit:
Merge pull request #25247 from owncloud/fed-unshare-fail

Remove a fed share from the local table before trying to notify the remote server

Commit:
Merge pull request #25129 from owncloud/remove-ie

Remove ie8+9 support

MorrisJobke committed:
Merge pull request #25252 from owncloud/authtoken-removetooltipondisconnect

Remove tooltip when disconnecting token

Commit:
Quickfix: do not lazy load auth mechanisms for ext storages

Some auth mechanisms like SessionCredentials need to register hooks
early, so they cannot be lazy loaded.

Commit:
Merge pull request #25258 from owncloud/integritycheck-whennotinstalled

Make code integrity check work when OC is not installed yet

Commit:
Merge pull request #25250 from owncloud/linkshare-includedeletewithuploadperms

Add explicit delete permission to link shares

ChristophWurst committed:
Merge pull request #25263 from owncloud/decryptall-keepencryptionenabledforsingleuser

Keep encryption enabled if decrypting for single user

Commit:
Merge pull request #25261 from owncloud/app-password-login-name

show which login name to use for the new app password

Commit:
Merge pull request #25237 from owncloud/search-filelistnextpageresults

Prerender file list pages to include search results

Commit:
Merge pull request #25273 from owncloud/ext-fixsessioncredentialsnolazyload

Quickfix: do not lazy load auth mechanisms for ext storages

ChristophWurst committed:
prevent users from deleting their own session token

icewind1991 committed:
Log reason why we removed the fed share

georgehrke committed:
fix a ImageExportPlugin Test (#25215)

icewind1991 committed:
handle unavailable fed shares while testing for availability (#25277)

* More explicit http status codes

* handle unavailable fed shares while testing for availability

ChristophWurst committed:
Login hooks (#25260)

* fix login hooks

* adjust user session tests

* fix login return value of successful token logins

* trigger preLogin hook earlier; extract method 'loginWithPassword'

* call postLogin hook earlier; add PHPDoc

icewind1991 committed:
Fix getting the certificate bundle for dav external storage (#25274)

* Fix getting the certificate bundle for dav external storages

* Log the original exception in dav external storage

Commit:
[tx-robot] updated from transifex

Commit:
Hide revert button when no permission to revert

Commit:
[tx-robot] updated from transifex

schiessle committed:
get only vcards which match both the address book id and the vcard uri (#25294)

Commit:
Merge pull request #25288 from owncloud/fix-versionrevertperms

Hide revert button when no permission to revert

DeepDiver1975 committed:
Various database migration fixes (#25209)

* String columns with a length higher then 4000 are converted into a CLOB columns automagically - we have to respect this when migrating

* Adding schema migration tests to prevent unnecessary and non-sense migration steps
Fix Oracle autoincrement and unsigned handling

* Fix sqlite integer type for autoincrement

* Use lower case table names - fixes pg

* Fix postgres with default -1 - this only affect pg 9.4 servers - 9.5 seems to work fine

Commit:
9.1.0 RC 1

Commit:
fix version string

Henni committed:
adjusts tests to removal of svg fallback

Henni committed:
always use local karma

Commit:
[tx-robot] updated from transifex

Commit:
Prevent infinite loop in search auto-nextpage

When loading the next page of search results, make sure that the loop
can end if there are no more elements in case the total doesn't match.

Also added a check to avoid recomputing the search results whenever the
setFilter() is called with the same value. This happens when navigating
away to another folder, the search field gets cleared automatically and
it calls FileList.setFilter('').

icewind1991 committed:
hide hidden parameters from list backend/auth parameters

icewind1991 committed:
show configuration options for authentication backends while listing storage

Fixes #22447

Commit:
Merge pull request #25310 from owncloud/search-preventinfiniteloop

Prevent infinite loop in search auto-nextpage

Commit:
Merge pull request #25304 from owncloud/local-karma

Always use local karma

Commit:
[tx-robot] updated from transifex

Commit:
Filelist change dir auto-prepend slash

Prepend a slash to directories in case it was missing since many places
assume that it's there.

Commit:
Workaround to check htaccess in case of redirects

In some setups, the web server will redirect any call to "data/" to the
main page. This causes the XHR to return the 200 HTTP status code and
the body contains the HTML page of the main page / files app.

This fix improves the htaccess failure detection by adding a known
string inside the test file "htaccesstest.txt". If we are able to find
this string, it means that the web server didn't block access to that
file.

tcitworld committed:
Add all properties while creating a subscription (#25318)

Fixes #24469

Commit:
Merge pull request #25276 from owncloud/delete-own-session-token

prevent users from deleting their own session token

Commit:
Merge pull request #25262 from owncloud/fed-sharing-error

Only save federated share after remote server is notified

Commit:
Fix overwriting folders on rename with SFTP

This aligns the behavior with other storages and also fixes the failing
unit test testRenameOverWriteDirectory

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

butonic committed:
Cap LDAP user cache (#25323)

felixboehm committed:
check if renamed user is still valid by reapplying the ldap filter (#25338)

nickvergessen committed:
Change nickvergessen mailmap entry (#25349)

Commit:
Merge pull request #25314 from owncloud/files_external-backends-config

show configuration options for authentication backends

Commit:
Merge pull request #25253 from owncloud/users-fixotherquotadropdown

Fix "Other" value handling in quota dropdown in users page

MorrisJobke committed:
add unit tests

MorrisJobke committed:
Merge pull request #25240 from owncloud/remove-svg

Remove SVG fallback

Commit:
Added storage tests for fopen with special chars

This makes it possible to test special chars with unit tests.
There is already a test for directories but there was none for file
names.

VicDeo committed:
Bypass upgrade page when occ controller is requested

Commit:
[tx-robot] updated from transifex

DeepDiver1975 committed:
fix swift primary object store test (#25281)

* Wait for socket to be open

* Fix call on null

* Allow DB access for MountProviderTest

Makes unit tests pass when using object store, since their FS access is
actually oc_filecache DB access. It is currently not possible to mock
or bypass the logic from "SharedMount::verifyMountPoint()" triggered by
this test.

Commit:
Add integration test for sharing with group, then user in group

Add integration test for the use case where a group share exists and
then the same owner creates a direct share to a user in that group.

Commit:
Merge pull request #25344 from owncloud/fix-ldap-check-user

check if renamed user is still valid by reapplying the ldap filter

Commit:
Merge pull request #25356 from owncloud/checkupgrade-bypass

Bypass upgrade page when occ controller is requested

Commit:
[tx-robot] updated from transifex

cdamken committed:
Upgrading the log level From Debug to INFO to be consistent with files_trashbin (#25367)

Commit:
Ignore invalid paths in the JS file list (#25368)

Commit:
Merge pull request #25325 from owncloud/filelist-prependslash

Filelist change dir auto-prepend slash

Commit:
Merge pull request #25361 from owncloud/sharing-inttestforusershareaftergroupshare

Add integration test for sharing with group, then user in group

Commit:
Additional perm check in Webdav

Commit:
Use named parameter instead of direct value for system tags search param (#25380)

Commit:
Set content type when downloading log file to force download in some browsers (#25382)

Commit:
[tx-robot] updated from transifex

MorrisJobke committed:
Simplify test code and don't show anything if no date is set

jvillafanez committed:
Change png icons to svg

Commit:
Fix js unit test in filelist spec

Commit:
Merge pull request #25405 from owncloud/ldap_png_to_svg

Change png icons to svg

Commit:
Merge pull request #25407 from owncloud/filelist-fixjsunittest

Fix js unit test in filelist spec

Commit:
Merge pull request #19513 from owncloud/add-expiration-date-row

Add expiration date to share by link view

Commit:
[tx-robot] updated from transifex

guruz committed:
Locking: Add testcases for lock length #25376

DeepDiver1975 committed:
Add clipboard button to public share link

DeepDiver1975 committed:
Use an icon to set the email instead of a button with string

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
Cast share id to string (#25402)

Commit:
Merge pull request #25331 from owncloud/htaccesstest-redirectworkaround

Workaround to check htaccess in case of redirects

RealRancor committed:
Exclude owncloud-news-gitter-bot from mention bot (#25436)

Commit:
Fix setup check unit test (#25439)

SergioBertolinSG committed:
Modified integration tests

Commit:
Add integration tests for merging received shares

Commit:
Improved share grouping readability + fixed test

Commit:
[tx-robot] updated from transifex

Commit:
Merge pull request #25388 from owncloud/webdav-copypermissions

Additional perm check in Webdav

Commit:
Merge pull request #25352 from owncloud/storage-fopenspecialchars

Added storage tests for fopen with special chars

Commit:
Merge pull request #25340 from owncloud/sftp-renameoverwritedir

Fix overwriting folders on rename with SFTP

Commit:
Add repair step for unmerged shares (WIP)

DeepDiver1975 committed:
Adding certificate revocation list and validate if the app certificate is revoked

Commit:
[tx-robot] updated from transifex

yarwest committed:
Added fix for issue #25332 (#25355)

Make sure that the newly entered password is not the same as the
existing password. This is done using a simple if statement comparing
already defined variables for the newly entered and the old password. If
the variables are the same a fitting error will be passed.

DeepDiver1975 committed:
Check integrity of a signed app in any case on installation

Commit:
Added more tests for sharing's MountProvider

Commit:
Merge pull request #25459 from owncloud/revoke-app-dev-cert

Adding certificate revocation list and validate if the app certificat…

Commit:
Group incoming shares for resharing in JS

Commit:
[tx-robot] updated from transifex

Commit:
Adjust repair version check for unmerged shares

DeepDiver1975 committed:
Change load order of auth backends so that we can throw an exception in OCA\DAV\Connector\Sabre\Auth - fixes #25362 (#25476)

Commit:
[tx-robot] updated from transifex

SergioBertolinSG committed:
Added custom exception as non fatal (#25486)

Commit:
Merge pull request #25418 from owncloud/use-clipboard-api

[9.2] Add clipboard button to public share link

dragotin committed:
Show app name in error message if app could not be loaded. (#25441)

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

DeepDiver1975 committed:
Test jenkins pipeline (#25401)

DeepDiver1975 committed:
Use explode() instead of split() - fixes #25483 (#25488)

Commit:
[tx-robot] updated from transifex

DeepDiver1975 committed:
Kill the old ftp storage implementation - refs #19941 (#25313)

Commit:
Increase version to 9.2 (#25510)

DeepDiver1975 committed:
Use phantomjs-prebuilt as warning is telling us

Commit:
[tx-robot] updated from transifex

Commit:
Merge pull request #25516 from owncloud/phantomjs-prebuilt

Use phantomjs-prebuilt as warning is telling us

Commit:
Merge pull request #25113 from owncloud/group_shares

Group shares with same source and target

Commit:
Adjust versions in info.xml to 9.2 (#25537)

Commit:
[tx-robot] updated from transifex

DeepDiver1975 committed:
Update copyright (#25518)

* Copyright is with ownCloud GmbH now

* Recreate all license headers

DeepDiver1975 committed:
Adding smb files external tests against a windows server (#25523)

aaronjwood committed:
Escape special characters (#25429)

* Escape LIKE parameter

* Escape LIKE parameter

* Escape LIKE parameter

* Escape LIKE parameter

* Escape LIKE parameter

* Use correct method in the AbstractMapping class

* Change the getNamesBySearch method so that input can be properly escaped while still supporting matches

* Don't escape hardcoded wildcard

jvillafanez committed:
Fix file permissions for SMB (read-only folders will be writeable) (#25301)

* Fix file permissions for SMB (read-only folders will be writeable)

* Read-only folders won't be deletable

* Added comment for the read-only behaviour for folders

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
Fix RepairUnmergedShares to not skip valid repair cases

The repair step was a bit overeager to skip repairing so it missed the
case where a group share exists without subshares but with an
additional direct user share.

Commit:
Merge pull request #25567 from owncloud/group-share-repairstepfix

Fix RepairUnmergedShares to not skip valid repair cases

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

DeepDiver1975 committed:
Especially with coverage enabled execution times are pretty high (#25517)

Commit:
Simplify testPermissionMovedGroupShare (#25573)

Reduces significantly the testing time but still cover the permission
change

DeepDiver1975 committed:
Fixes undefined two-factor-providers (#25605)

DeepDiver1975 committed:
Redirect to default page after login - fixes #25562 (#25600)

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
Creating second share to user who renamed received folder creates a second entry (#25568)

* Add integration tests for double shares with rename in between

* Make share target consistent when grouping group share with user share

In some situations, a group share is created before a user share, and
the recipient renamed the received share before the latter is created.
In this situation, the "file_target" was already modified and the second
created share must align to the already renamed share.

To achieve this, the MountProvider now groups only by "item_source"
value and sorts by share time. This makes it so that the least recent
share is selected as super-share and its "file_target" value is then
adjusted in all grouped shares.

This fixes the issue where this situation would have different
"file_target" values resulting in two shared folders appearing instead
of one.

Commit:
Prevent setting email and triggering events at login time (#25553)

Whenever an LDAP user also has an email address defined in LDAP, the
LDAP code will try and update the email address of the locally known
user. This happens at login time or every time the user's LDAP
attributes are processed.

There is code listening to the email setting hook which updates the
system address book, which also will trigger FS setup due to avatars
and other things.

This fix only sets the email address when really necessary.

Commit:
Fix misleading MySQL DB creation error (#25485)

Whenever the GRANT ALL failed, it used to display "Database creation
failed" which is incorrect. It's only the privleges setting that failed.

This moves the privilege setting message to DEBUG and makes it more
precise.

jvillafanez committed:
Close session fast when doing a test to prevent blocking the next (#25249)

requests

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

nickvergessen committed:
Allow to change user preferences via occ (#25083)

* Add a occ command to list/get user preferences

* Allow setting values

* Allow deleting a setting

* Add tests for checkInput()

* Make sure the exception is catched

* Add tests for set/get/delete/list

* Fix descriptions

DeepDiver1975 committed:
Adding quota plugin to new dav endpoint (#25615)

* Adding quota plugin to new dav endpoint

* Added integrated test failing in old endpoint

* Added 0B quota test

DeepDiver1975 committed:
Ensure the user exists before calling a method on it - fixes #24751

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

RealRancor committed:
forum.owncloud.org -> central.owncloud.org (#25644)

Commit:
[tx-robot] updated from transifex

nickvergessen committed:
Move Share backends to PSR-4 instead of using class path (#24941)

DeepDiver1975 committed:
Load the app before executing it's repair steps (#25650)

SergioBertolinSG committed:
Integration test for 25582 (#25623)

* Added integration test

* A shared storage is never a home storage - fixes #25582

Commit:
[tx-robot] updated from transifex

jvillafanez committed:
Hide the LDAP password in the client side

Connection checks will be done by using the configuration id, with the
stored password. LDAP password won't be sent to the client.

DeepDiver1975 committed:
Next step jenkinsfile (#25622)

* Adding timestamper and evaluation of test results even in case of failure

* Adding build timeout

* use fixed value 120 minutes as timeout for each test executing for now

* Terminate the build as soon as test execution fails

* Adjust external testing as well

* Finalize use of executeAndReport

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

DeepDiver1975 committed:
Add missing array element - fixes #25714

Commit:
Merge pull request #25472 from owncloud/92_testLongLock

Locking: Add testcases for lock length #25376

Commit:
Merge pull request #25717 from owncloud/fix-undefined-offset-in-login-controller

Fix undefined offset in login controller

Commit:
[tx-robot] updated from transifex

DeepDiver1975 committed:
Remove 9.1 migration steps in app dav (#25654)

VicDeo committed:
Respect theme for external folder icon (#25487) (#25712)

See issue #25461. 
When using a theme with alternative filetype icons `OCA.External.StatusManager.Utils.getIconRoute` should respect that, rather than hard coding the default icon.

Note this change does not affect windows_network_drive and sharepoint icons, which appear not to be easily themeable.

DeepDiver1975 committed:
getJailedPath expects $path to have a trailing /  (#25703)

* getJailedPath expects $path to have a trailing / - fixes #25464

* Adding test case for getPathById including a jailed cache where root is just empty

kavishdahekar committed:
Fix for issue #20050 (#25735)

Minor change in width of input box.

nickvergessen committed:
Allow downgrades of maintenance accross vendors (#25577)

Caffe1neAdd1ct committed:
Fix for bug in PHP FPM through both ProxyPassMatch and SetHandler where
 the global variable SERVER SCRIPT_NAME isn't set correctly on apache 
 2.4.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/core?pullRequest=25747) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/core?pullRequest=25747) before we can accept your contribution.<br/>

DeepDiver1975 commented on lib/private/AppFramework/Http/Request.php:
```diff
@@ -770,5 +770,18 @@ private function getOverwriteHost() {
+        if (strpos($this->server['SCRIPT_NAME'], '.php') !== false) {
+            $name = DIRECTORY_SEPARATOR . basename($this->server['SCRIPT_FILENAME']);
+        }
+        return $name;
```
$name will be undefined in case the if statement is not evaluated to true :-1:

Caffe1neAdd1ct replied:
Thanks @DeepDiver1975 I'll look at that now.

DeepDiver1975 commented:
@Caffe1neAdd1ct thanks for your pull request - please reference bugs this is going to fix - THX

RealRancor commented:
Workaround for a Bug in PHP which affects only Apache2 running with PHP-FPM: https://bugs.php.net/bug.php?id=65641

This might replace: https://github.com/owncloud/core/pull/17491

Caffe1neAdd1ct commented:
Potential Bug References:

https://github.com/owncloud/core/issues/22411
https://github.com/owncloud/core/issues/21871
https://github.com/owncloud/core/issues/7719

wpoely86 commented:
Backport request for RHEL 7: https://bugzilla.redhat.com/show_bug.cgi?id=1248077

Caffe1neAdd1ct committed:
Fixed undefined variable notice when script name would not have .php 
in the string.

wpoely86 commented:
@Caffe1neAdd1ct Can you show the apache config you're using for this? It doesn't seem to work for me with 9.0.4 on CentOS7.

Caffe1neAdd1ct commented:
Manual tests i'm going to run through :

- [ ]  Test webapp with apache ProxyPassMatch FPM
- [ ]  Test webapp with apache SetHandler FPM
- [ ] Test webapp with apache mod_php
- [ ] Test webapp with nginx fpm setup

- [ ]  Test sync client with apache ProxyPassMatch FPM
- [ ]  Test sync client with apache SetHandler FPM
- [ ] Test sync client with apache mod_php
- [ ] Test sync client with nginx fpm setup

Caffe1neAdd1ct commented:
@wpoely86 Apache config on my servers:

```
<VirtualHost *:80>

    ServerName owncloud.castle
    ServerAdmin support@castle
    DocumentRoot /usr/share/webapps/owncloud/

    SetEnvIf Authorization "(.*)" HTTP_AUTHORIZATION=$1
    SetEnv HOME /usr/share/webapps/owncloud
    SetEnv HTTP_HOME /usr/share/webapps/owncloud
    
    DirectoryIndex index.php
    ProxyPassMatch ^/(.*\.php(/.*)?)$ fcgi://127.0.0.1:9001/usr/share/webapps/owncloud/$1

    <Directory "/usr/share/webapps/owncloud">
        AllowOverride All
        Require all granted
    </Directory>

    ErrorLog "/var/log/httpd/owncloud.error.log"
    CustomLog "/var/log/httpd/owncloud.access.log" combined

</VirtualHost>
```

Commit:
[tx-robot] updated from transifex

DeepDiver1975 committed:
In case of unit testing - just return the default services (#25749)

Caffe1neAdd1ct commented:
Looks like i have some tests to look at:

```

	Test Result (14 failures )
Test\AppFramework\Http\RequestTest::testGetPathInfoWithoutSetEnvGeneric.testGetPathInfoWithoutSetEnvGeneric with data set #0
Test\AppFramework\Http\RequestTest::testGetPathInfoWithoutSetEnvGeneric.testGetPathInfoWithoutSetEnvGeneric with data set #1
Test\AppFramework\Http\RequestTest::testGetPathInfoWithoutSetEnvGeneric.testGetPathInfoWithoutSetEnvGeneric with data set #2
Test\AppFramework\Http\RequestTest::testGetPathInfoWithoutSetEnvGeneric.testGetPathInfoWithoutSetEnvGeneric with data set #3
Test\AppFramework\Http\RequestTest::testGetPathInfoWithoutSetEnvGeneric.testGetPathInfoWithoutSetEnvGeneric with data set #4
Test\AppFramework\Http\RequestTest::testGetRawPathInfoWithoutSetEnvGeneric.testGetRawPathInfoWithoutSetEnvGeneric with data set #0
Test\AppFramework\Http\RequestTest::testGetRawPathInfoWithoutSetEnvGeneric.testGetRawPathInfoWithoutSetEnvGeneric with data set #1
Test\AppFramework\Http\RequestTest::testGetRawPathInfoWithoutSetEnvGeneric.testGetRawPathInfoWithoutSetEnvGeneric with data set #2
Test\AppFramework\Http\RequestTest::testGetRawPathInfoWithoutSetEnvGeneric.testGetRawPathInfoWithoutSetEnvGeneric with data set #3
Test\AppFramework\Http\RequestTest::testGetRawPathInfoWithoutSetEnvGeneric.testGetRawPathInfoWithoutSetEnvGeneric with data set #4
Show all failed tests >>>
```

I'll have a look into these plus making sure this change doesn't break other setups.

Caffe1neAdd1ct committed:
Fixed passing of script name to sabre splitpath method, this should be 
the raw script name with the path_info merge onto the end as expected.

Caffe1neAdd1ct committed:
Fixed failing tests due to new required server variable SCRIPT_FILENAME

Caffe1neAdd1ct committed:
Fixed re-introduction of login redirect loop and completed FIXME for 
removal of Sabre splitpath usage. Fixed tests to work with rewritten 
rawpath method.

DeepDiver1975 commented:
still 15 failing unit tests - https://ci.owncloud.org/job/ownCloud/job/core/job/PR-25747/2/testReport/

Caffe1neAdd1ct committed:
Fixed request uri having path before script name being processed by 
script name.

Caffe1neAdd1ct commented:
@DeepDiver1975 Working on them, running them again locally to find the issues through `./autotest.sh`

Caffe1neAdd1ct committed:
Fixed zero int return failing strpos check to make sure requesturi is 
only runnable through script name.

Caffe1neAdd1ct commented:
Still an issue with the test env for phpunit putting /core/ on the front of requests, having another look

butonic committed:
log class name, ID only is hard to debug (#25743)

butonic committed:
Skip version and trash expiry for users that never logged in (#25741)

Caffe1neAdd1ct committed:
Re-implemented raw path info function to take into account all different 
setups including php-fpm sethandler or proxypassmatch plus mod_php with
 or without subdriectory of core as per tests.

Caffe1neAdd1ct commented:
The 210 failing tests look like they are due to the tests running with a document root of the folder above core, meaning owncloud is running in a sub folder of core.

Is this a supported configuration?

Commit:
[tx-robot] updated from transifex

butonic committed:
use $userId instead of $user

butonic committed:
Unnecessary fully qualified names

butonic committed:
Unreachable statement

butonic committed:
Method is deprecated

butonic committed:
Type compatability

butonic committed:
missing PHPDoc

Caffe1neAdd1ct committed:
Merge origin/master

Caffe1neAdd1ct committed:
Added test data to cover token generate url.

Caffe1neAdd1ct committed:
Fixed to cater for all test data which overs a range of url formats 
with the fix for fpm if path_info has been appended to script_name.

Caffe1neAdd1ct committed:
Removed debug code.

Caffe1neAdd1ct commented:
Waiting for broken tests to be fixed in master, will pull the master branch when the tests pass and re-run the pull request branch.

DeepDiver1975 committed:
Fix discoverymanager tests (#25769)

* Allow testing of the DiscoveryManager

* Array sort order is of no relevance

Caffe1neAdd1ct committed:
Merge origin/master

Caffe1neAdd1ct commented:
Pulled master changes which fix 4 broken tests (d2e20213f5f5ad9ea25bd46311157747ba7d4d22) jenkins build scheduled.

Caffe1neAdd1ct commented:
Issue with integration tests setting up users and data through ocs cli.

Commit:
Merge pull request #25652 from owncloud/fix-getQuota-on-null

Ensure the user exists before calling a method on it

Commit:
Merge pull request #25767 from owncloud/polish-user-code

How to polish code, eg: OC\User

Commit:
Merge pull request #25702 from owncloud/ldap_hide_configured_password

Hide the LDAP password in the client side

Commit:
[tx-robot] updated from transifex

Commit:
Fix OR operands in SharedMountTest (#25778)

Caffe1neAdd1ct commented:
Issue located, script name fix is cutting off the /ocs/ subfolder from the script name. Writing tests and fixing issue hopefully today.

butonic committed:
remove dead code and unused imports (#25763)

Commit:
Store the shared propagator instance (#25509)

This instead of recreating it for every call.

ppaysant committed:
Settings sections id do not change with i18n (#25528)

Caffe1neAdd1ct committed:
Added test for ocs api path due to broken integration tests.

Caffe1neAdd1ct commented:
Unit tests and integration tests passing on my local development environment, pushed changes and waiting on Jenkins PR build.

Caffe1neAdd1ct commented:
PHPUnit tests have failed due to `An exception occured in driver: SQLSTATE[HY000] [1040] Too many connections`

This seems unrelated to the current changes and the MySQL PHPUnit tests are passing on my environment without any issues.

There were a lot of tests running when this PR branch was kicked off which may have caused this.

@DeepDiver1975 are you able to kick off this build again?

butonic committed:
lastInsertId requires the prefix (#25764)

DeepDiver1975 commented:
@Caffe1neAdd1ct nice jobs - thx - can I ask you to squash the commits? THX

Caffe1neAdd1ct commented:
@DeepDiver1975 i'm out at the moment, i think GitHub's UI automates this when merging in a pull request (https://help.github.com/articles/about-pull-request-merge-squashing/)

If not i can have a go Monday :+1:

butonic committed:
add conditional smb trace logging for debug purposes (#25758)

Commit:
[tx-robot] updated from transifex

Caffe1neAdd1ct committed:
Fixed rawpathinfo for paths with leading directories in script name.
This fixes calls to ocs failing on the integration tests.

Caffe1neAdd1ct commented:
New dev env had a localhost git author set, i've amended the commit however cla is still showing the non existent contributor as required to sign the cla.

Happened when squashing the commits via git cli rather than through NetBeans.
